Swift 4 Decodable pattern for basic queries

patrick-zippenfenig opened the pull request:
```swift
struct User: Decodable {
  let id: Int
  let user_name: String
  let age: Int?
}
let rows: [User] =  try conn.query("SELECT id, user_name FROM users WHERE id = 123")
```
See https://github.com/novi/mysql-swift/issues/58. QueryRowResultType still supported.

Commit:
* Decodable pattern for basic queries

patrick-zippenfenig committed:
Decode Data type for binary blobs

novi commented on Sources/MySQL/Result.swift:
```diff
@@ -142,3 +142,120 @@ public struct QueryRowResult {
+    }
+}
+
+struct RowKeyedDecodingContainer<K : CodingKey> : KeyedDecodingContainerProtocol {
```
@patrick-zippenfenig We could change `struct`  to `fileprivate struct`.

novi commented:
Thanks for the pull request.
The API design looks good.

Could you write a test for this decodable support?

patrick-zippenfenig committed:
- Decode Date
- 2x Decodable Unit tests
Note: AutoincrementID is not yet decodable.... need to write AutoincrementID: Decodeable extension

patrick-zippenfenig commented:
@novi Basic tests seem to be fine. I implemented a first version of "Encodable" locally. It's a little bit more complicated then Decodable. I will start another PR for encoding

novi commented:
@patrick-zippenfenig Thanks and the test is fine.
I noticed that `AutoincrementID` is not yet decodable and `Data` (Blob) test is absent.
But I will merge this PR for now. We could open another PR for that.
